libafl_targets/src/sancov_cmp.rs

```diff
@@ -23,3 +29,67 @@ extern "C" {
     pub fn __sanitizer_cov_trace_switch(val: u64, cases: *const u64);
 
 }
+
+#[no_mangle]
+unsafe extern "C" fn __sanitizer_cov_pcs_init(pcs_beg: *const usize, pcs_end: *const usize) {
+    // "The Unsafe Code Guidelines also notably defines that usize and isize are respectively compatible with uintptr_t and intptr_t defined in C."
+    assert!(
+        PCS_BEG.is_null(),
+        "__sanitizer_cov_pcs_init can be called only once."
+    );
+    assert!(
+        PCS_END.is_null(),
+        "__sanitizer_cov_pcs_init can be called only once."
+    );
+
+    PCS_BEG = pcs_beg;
+    PCS_END = pcs_end;
+}
+
+/// An entry to the sanitizer_cov pc_table
+#[repr(C, packed)]
+#[derive(Debug, PartialEq, Eq)]
+pub struct PcTableEntry {
+    addr: usize,
+    flags: usize,
+}
+
+impl PcTableEntry {
+    /// Returns whether the PC corresponds to a function entry point.
+    pub fn is_function_entry(&self) -> bool {
+        self.flags == 0x1
+    }
+
+    /// Returns the address associated with this PC.
+    pub fn addr(&self) -> usize {
+        self.addr
+    }
+}
+
+/// Returns a slice containing the PC table.
+pub fn sanitizer_cov_pc_table() -> Option<&'static [PcTableEntry]> {
+    unsafe {
```

Comment:
Maybe add a safety doc here that once `PCS_BEG` has been initialized, it'll never be written to again, so no TOCTU is possible

Reply:
I am not entirely sure this is true. `__sanitizer_cov_pcs_init` would be called once for each instrumented DSO, so we could have multiple calls in a (badly?) compiled binary. I think libafl is typically used with only one statically linked binary, but nothing is forbidding this code to be misused, I think.